### PR TITLE
Update to junit5/assert4j

### LINKED ISSUE #127: Update simple-task and spring-batch examples to use junit5/assert4j to agree with what gets generated from initializr

### PR DESCRIPTION
**Pull request by markpollack**
Fixes #127

### CODE REVIEW

File: dataflow-website/batch-developer-guides/batch/batchsamples/pom.xml [hunk 1]
```diff
@@ -34,13 +34,13 @@
...
 	<dependencyManagement>
 		<dependencies>
-			<dependency>
-				<groupId>org.springframework.cloud</groupId>
-				<artifactId>spring-cloud-task-dependencies</artifactId>
-				<version>2.2.0.BUILD-SNAPSHOT</version>
-				<type>pom</type>
-				<scope>import</scope>
-			</dependency>
+<!--			<dependency>-->
```
Comment (jvalkeal) [inline dataflow-website/batch-developer-guides/batch/batchsamples/pom.xml]: Can be removed?

File: dataflow-website/batch-developer-guides/batch/batchsamples/billrun/pom.xml [hunk 2]
```diff
@@ -45,6 +45,12 @@
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-test</artifactId>
 			<scope>test</scope>
+			<exclusions>
```
Comment (jvalkeal) [inline dataflow-website/batch-developer-guides/batch/batchsamples/billrun/pom.xml]: I don't really see no point to exclude these unless we want to make sure there's no junit4 tests?

Comment (jvalkeal) [reply dataflow-website/batch-developer-guides/batch/batchsamples/billrun/pom.xml]: Didn't realise until just now that initializr does this same so now I understand why you did it.

File: dataflow-website/batch-developer-guides/batch/batchsamples/pom.xml (hunk 1 above)
Comment (markpollack) [reply dataflow-website/batch-developer-guides/batch/batchsamples/pom.xml]: sure!

**Comment by markpollack** [comment]
addressed PR comments and merged in f205626b4f3523c338578838416a2bc29a80fb03

**Pull request closed without merging**